Foundation/JSONSerialization.swift

```diff
@@ -812,6 +812,35 @@ private struct JSONReader {
         0x2E, 0x2D, 0x2B, 0x45, 0x65, // . - + E e
     ]
     func parseNumber(_ input: Index, options opt: JSONSerialization.ReadingOptions) throws -> (Any, Index)? {
+       func castInt64(_ value: Int64) -> Any {
+            if Int.bitWidth == 64 {
+                return Int(value)
+            } else {
+                if let result = Int(exactly: value) {
+                    return result
+                }
+
+                return value
+            }
+        }
+
+        func castUInt64(_ value: UInt64) -> Any {
+            if let result = Int(exactly: value) {
+                return result
+            }
+
+            if Int.bitWidth == 64 {
```

Comment:
Should this be `UInt.bitWidth`?

Reply:
Maybe. But I think it is impossible to Int.bitWidth != UInt.bitWidth